commons-math-legacy/src/main/java/org/apache/commons/math4/legacy/stat/regression/MatrixOperations.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.math4.legacy.stat.regression;
+
+import org.apache.commons.math4.legacy.linear.Array2DRowRealMatrix;
+import org.apache.commons.math4.legacy.linear.QRDecomposition;
+import org.apache.commons.math4.legacy.linear.RealMatrix;
+import org.apache.commons.math4.legacy.linear.RealVector;
+
+/**
+ * Helper class encapsulating matrix operations for OLSMultipleLinearRegression.
+ */
+class MatrixOperations {
```

Comment:
It is unclear why we are moving the methods to this class. The class should be final; the methods should be static and package-private, i.e. the calling class should not have to maintain an instance to invoke the methods.

All this change has done is duplicate the comment from the original location and moved the code. Thus the code becomes detached from the original comment. If this subsequently changes over time then this can introduce documentation/implementation mismatch.